### PR TITLE
Third party docs fixes/updates

### PR DESCRIPTION
**Pull request by jbednar**
The link in master from the homepage to the "Third party packages" page was broken, because it was still directing people to mpl_toolkits, even though the contents have moved to thirdpartypackages/.  The organization of the third party packages page itself was also hard to follow, because it listed several smaller and miscellaneous packages before substantial packages like seaborn.  

Reorganized to first discuss large projects providing alternative APIs for matplotlib, including the new HoloViews package, then large geographic plotting packages, and then the remaining smaller packages (which seem difficult to categorize into meaningful groups).  Also added images for seaborn and ggplot, so that people have a better idea of what they offer, and to make the page be more balanced visually.  

The resulting web pages seem to build ok; e.g.:

http://homepages.inf.ed.ac.uk/jbednar/mpl/thirdpartypackages/index.html

http://homepages.inf.ed.ac.uk/jbednar/mpl/index.html

I've applied this pull request to the master, because it is meant to update the most current copy of the web site, but let me know if a change to the docs like this should have been against another branch.

### CODE REVIEW

**Comment by mdboom** [comment]
:+1:  I'd be in favor of merging this to 1.5.x-docs, just so it gets deployed sooner, but there's nothing for you to do -- we can backport.

**Comment by tacaswell** [comment]
I am a bit concerned that our link-checker did not find the broken link.

File: doc/thirdpartypackages/index.rst [hunk 1]
```diff
@@ -5,75 +5,105 @@
...
-ggplot
-======
 
-`ggplot <https://github.com/yhat/ggplot>`_ is a port of the R ggplot2
-to python based on matplotlib.
+
+Cartopy
+=======
+
+An alternative mapping library written for matplotlib ``v1.2`` and beyond.
```
Comment (QuLogic) [inline doc/thirdpartypackages/index.rst]: While you're here, can you change this to 1.3? The next release will drop support for 1.2.

**Comment by jenshnielsen** [comment]
@tacaswell The link is not broken It is just wrong

**Comment by jenshnielsen** [comment]
@mdboom This will not backport directly since my doc changes splitting these are not backported as is

File: doc/thirdpartypackages/index.rst (hunk 1 above)
Comment (jbednar) [inline doc/thirdpartypackages/index.rst]: I've just deleted that line altogether. I don't think matplotlib's website needs to keep track of that information; people can find out about supported versions from cartopy's website, which cartopy's maintainers keep up to date...

**Comment by jbednar** [comment]
Sorry to be ambiguous -- yes, the link was not really broken, just out of date.  It was pointing to a real page, just not the correct one.

I'd be happy to make a similar (but incompatible due to the page splitting) pull request for the 1.5.x branch; just let me know.

**Comment by tacaswell** [comment]
@jbednar That would be great, thanks.

**Comment by jbednar** [comment]
Ok, I pulled out the novel content and put it into a separate PR for the docs branch (#6106).  I didn't attempt to re-do the reorganization that @jenshnielsen did; I just added the new bits in the locations they already were.

**Comment by tacaswell** [comment]
I have merged #6106 and when will do the merge cascade + doc rebuild later tonight.  This PR will likely require a rebase after that.

**Comment by tacaswell** [comment]
@jbednar This needs to be re-based.

**Comment by jbednar** [comment]
I rebased, but it looks like a mess now.  I presume I should just checkout master again and replay my original changes for a new PR?

**Comment by tacaswell** [comment]
Or just force-push that new branch to this PR.  

It looks like you did the rebase backwards (master onto your branch, not your branch onto master)?

**Comment by jbednar** [comment]
I didn't want to try to figure out those particular git mysteries, so I just built a new PR. :-/
